plugin/editorconfig.vim

```diff
@@ -209,15 +209,9 @@ function! s:UseConfigFiles(from_autocmd) abort " Apply config to the current buf
     " from_autocmd is truthy if called from an autocmd, falsy otherwise.
 
     " Get the properties of the buffer we are working on
-    if a:from_autocmd
-        let l:bufnr = str2nr(expand('<abuf>'))
-        let l:buffer_name = expand('<afile>:p')
-        let l:buffer_path = expand('<afile>:p:h')
-    else
-        let l:bufnr = bufnr('%')
-        let l:buffer_name = expand('%:p')
-        let l:buffer_path = expand('%:p:h')
-    endif
+    let l:bufnr = a:from_autocmd ? str2nr(expand('<abuf>')) : bufnr('%')
+    let l:buffer_name = expand('#'.l:bufnr.':p')
```

Comment:
My concern here is that, if `a:from_autocmd` is true, will a [No Name] buffer name cause other unexpected behaviors?

@cxw42 What do you think?

Reply:
As far as I know, `expand('#n')` does not raise error. If buffer `n` is a `[No Name]` buffer, `expand('#n:p')` is empty string and `expand('#n:p:h')` is cwd.

Reply:
Perfect! Looks like everything will be working fine. Thanks for the PR!